addon-test-support/@ember/test-helpers/dom/click.ts

```diff
@@ -82,12 +82,12 @@ export default function click(target: Target, options: MouseEventInit = {}): Pro
       throw new Error(`Element not found when calling \`click('${target}')\`.`);
     }
 
-    let isDisabledFormControl = isFormControl(element) && element.disabled;
-
-    if (!isDisabledFormControl) {
-      __click__(element, options);
+    if (isFormControl(element) && element.disabled) {
+      throw new Error(`Can not \`click\` disabled ${element}`);
```

Comment:
I'm not sure if we should throw an error for this. A real user is absolutely able to click on a disabled element, the difference though is that nothing will/should happen because the element is disabled.

Reply:
The intention of the PR is to disallow any interactions over disabled form elements. It seems the most effective to achieve it via throwing in the [focus](https://github.com/emberjs/ember-test-helpers/pull/681#issuecomment-557273957). This `throw` statement in `click` has been put here just for a better error message. If remove it, it'd [fail](https://github.com/emberjs/ember-test-helpers/pull/778/commits/714f0aa997307646e66ef4487228161c2937ba6a#diff-6fa0813ed7094e48d24ed8d3eaacd476R15) later in the `__focus__` util anyways.

>  A real user is absolutely able to click on a disabled element

yes, I agree about the user perspective. However, test env seems to be a bit different. If we don't fail as fast as we can, tracking the source of invalid test results can be more difficult then it could be if we were more strict.
In tests, we usually have all the state under control, and before clicking smth, we, as test authors, should likely know a desirable state for the button. 

For instance, if we have a button which is responsible for some computation, and we click it while it's disabled(because of some bug), we may have a failing `assert` for the expected computation value. And since we don't fail fast, the more distance between the action and assert, the more difficult to track the source of the issue. I believe test helpers could become more helpful here, by letting the user know if he tries to interact w/ something which is not supposed to be interactable.

Reply:
There are 2 possible reasons I can see, why someone clicks on a disabled element:
 - they intentionally did want to check if smth doesn't occur after click. In this case, the question is why not to just check for `disabled` value? And it also encourages the fail-late approach which I've tried to describe in the previous comment.
 -  the button should not have been disabled, but disabled because of some code issue. In this case we fall again in a "fail late" thing. And there is even a probability to have a false-positive test results if some assert is missing.

Curious if there are any other use cases for clicking disabled elements in tests? 🤔

Reply:
I agree with @Turbo87 - it shouldn't throw, just do nothing.

Reply:
I'd be happy to agree as well. But don't see a good reason for that. The only one which is raised:

> A real user is absolutely able to click on a disabled element, the difference though is that nothing will/should happen because the element is disabled.

however, in addition to my previous arguments, I don't find this aligned with some of the current behaviors. For instance, we currently fail on attempt to focus non-focusable elements https://github.com/emberjs/ember-test-helpers/blob/978f1b2f1c5314c6fcde6e30499ff90ba57feb95/addon-test-support/%40ember/test-helpers/dom/focus.ts#L70-L72
while in reality a user can absolutely try to focus on a disabled element or smth else w/o any success or exceptions in UI. 

Or, we  fail on attempt to fill-in non-editable elements https://github.com/emberjs/ember-test-helpers/blob/978f1b2f1c5314c6fcde6e30499ff90ba57feb95/addon-test-support/%40ember/test-helpers/dom/fill-in.ts#L40-L42

As far as I see, these all are just examples of fast feedback for the test helpers user, which allows restrict he/she from doing pointless things.

Reply:
In general, I agree with @ro0gr here. The general purpose helpers we provide here are intended to be demonstrating the functionality of the application, allowing `click('.some-disabled-thing')` to silently do nothing seems almost certainly to be an actual logic / error condition.